### PR TITLE
[DataObject] don't increase the version count when versioning is disabled

### PR DESCRIPTION
**Pull request by fabfischer**
## Changes in this pull request  
I noticed that the version count was increased, even with versioning disabled. the small fix/check honors such setting now:
```
if (Model\Version::$disabled === false) {
    $this->setVersionCount($this->getDao()->getVersionCountForUpdate() + 1);
}
```

## Additional info  
fixed one minor typo too

### CODE REVIEW

File: models/Element/AbstractElement.php [hunk 1]
```diff
@@ -49,7 +49,9 @@ abstract class AbstractElement extends Model\AbstractModel implements ElementInt
      */
     protected function updateModificationInfos()
     {
-        $this->setVersionCount($this->getDao()->getVersionCountForUpdate() + 1);
+        if (Model\Version::$disabled === false) {
```
Comment (brusch) [inline models/Element/AbstractElement.php]: I'd prefer introducing a static getter for that. Can you change the PR accordingly? Thanks in advance!

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/Element/AbstractElement.php (hunk 1 above)
Comment (fabfischer) [reply models/Element/AbstractElement.php]: Hi, 
thanks for your feedback. I hope I understood you correctly - I added two static methods that tell you if versioning is enabled or not:

```php
# models/Element/AbstractElement.php
if (Model\Version::isEnabled() === true) {
    $this->setVersionCount($this->getDao()->getVersionCountForUpdate() + 1);
}
```

```php
# models/Version.php
/**
* @return bool
*/
public static function isDisabled(): bool
{
    return (self::$disabled === true);
}

/**
* @return bool
*/
public static function isEnabled(): bool
{
    return (self::$disabled === false);
}
```

File: models/Version.php [hunk 2]
```diff
@@ -165,6 +165,22 @@ public static function enable()
...
     }
 
+    /**
+     * @return bool
+     */
+    public static function isDisabled(): bool
+    {
+        return (self::$disabled === true);
+    }
+
```
Comment (brusch) [inline models/Version.php]:
```suggestion
```

Comment (brusch) [reply models/Version.php]: `isEnabled` should be sufficient, can be used with `!`.

File: models/Version.php [hunk 3]
```diff
@@ -165,6 +165,14 @@ public static function enable()
         self::$disabled = false;
     }
 
+    /**
+     * @return bool
+     */
+    public static function isEnabled(): bool
+    {
+        return (self::$disabled === false);
```
Comment (brusch) [inline models/Version.php]:
```suggestion
        return !self::$disabled;
```

**Comment by brusch** [comment]
Thanks! 👍

**Pull request merged**